Add guide on contributing to the docs (#867)
meilisearch/documentation: learn/contributing/contributing_to_docs.md

# Documentation

This documentation website is hosted in a [public GitHub repository](https://github.com/meilisearch/documentation). It is built with [VuePress](https://vuepress.github.io), written in [MarkDown](https://www.markdownguide.org/getting-started/), and deployed on [Netlify](https://www.netlify.com).

## Our Documentation Philosophy

Our documentation aims to be:

- **Efficient**. We don't want to waste anyone's time.

> - **Simple**. Reading the documentation shouldn't require native English or an advanced degree.

Author: I was looking at this the other day and realized that it should be "accessible" :)

- **Thorough**. Behavior that isn't documented or is documented incorrectly, we consider broken.
- **Open source**. This is a resource by MeiliSearch users, for MeiliSearch users. [We welcome outside contributions](#how-to-contribute).

## How to Contribute?

### Issues

On the MeiliSearch documentation team, [GitHub Issues](https://github.com/meilisearch/documentation/issues/new/choose) are the primary tool we use to track tasks. They're like elements in our to-do list. Here are some good uses for issues:

- A detailed bug report

> - A request for a new technical feature (e.g. dark mode, versioning, etc.)

Author: I'll look into it...

- A request for new content (e.g. a tutorial, guide, or explanation)
- A helpful suggestion for the future of the documentation

> We love issues, because they help us do our jobs better. Nine times out of ten, the most useful contribution isn't a flashy [pull request](#pull-requests) that needs to be read and reviewed—it's a simple GitHub issue that points out a problem and proposes a solution.

Author: I would like to keep some aspect of this message in there, because encouraging issues over PRs will make my job easier. Maybe I can rephrase it so I'm not dissing PRs?

```suggestion
We love issues, because they help us do our jobs better. Nine times out of ten, the most useful contribution is a simple GitHub issue that points out a problem and proposes a solution.
```

#### Creating Your First Issue

All you need to create an issue is a [GitHub account](https://github.com).

1. Log into your account.
2. Go to the MeiliSearch [Documentation repository](https://github.com/meilisearch/documentation).
3. Click on "Issues".
4. Click "New issue".
5. Fill in a short, descriptive title and longer summary of your problem or suggestion.
6. Click "Submit new issue".
7. Enjoy the feeling of a job well done! 🎉

### Pull Requests

> Sometimes, the problem is so small or particular that it would be faster for you to fix it yourself than explain it to us. In those cases, rather than create an [issue](#issues), you should make a [pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests).

Author: How about this:
```suggestion
Sometimes, it's faster to fix a problem yourself than explain it to us. In those cases, rather than create an [issue](#issues), you should make a [pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests).
```

Pull requests ("PRs" for short) are requests to integrate your changes into an official repository. The simplest way to create a PR on our documentation is by using the **"Edit this page"** link at the bottom left of every page. [See below](#creating-your-first-pr) for more detailed instructions.

Pull requests are particularly good for:

- Fixing a small error, such as a typo, broken link, or poor word choice.
- Making a change that you are especially qualified for (such as creating a guide for some software that you have mastered).
- **Solving an [existing issue](https://github.com/meilisearch/documentation/issues)**!

> If the PR you're thinking about making isn't covered by the above list, it might be best to first [create an issue](#issues) to confirm that it's worth your time (and that nobody else is working on it already).

Author: Something like: 

```suggestion
Unless your PR is really small, we recommend you first [create an issue](#issues) to confirm that it's worth your time, and that nobody else is working on it already.
```

#### Creating Your First PR

There are many ways to create pull requests, but we'll cover the simplest method for this documentation site. Feel free to try it on this page!

All you need to continue is a [GitHub account](https://github.com).

> 1. On the documentation page you'd like to edit, scroll to the bottom and click **"Edit this page"** on the left. This will take you to GitHub.

Author: I think we want to keep these steps as simple as possible since it's for total beginners (plus the guide is designed for fixing typos and other small things, and those are unlikely to be sitting around as non-merged PRs 😄 )

4. Scroll down to where it says `Propose changes`. In the first text box, write a short description (e.g. "Fix typo"). In the second text box, write a longer description of your changes if necessary.
5. Click `Propose changes` to continue. You should see a page that says `Comparing changes`. This screen is prompting you to create a pull request.
6. Make sure the base repository is `meilisearch/documentation` and the base is `master`. Don't worry about the other two boxes—they automatically target the changes you just made.
7. Below, you can see the changes you made compared to the master branch (what's currently published on docs.meilisearch.com). When you're ready, click `Create pull request`.
8. **Congrats, you made your first PR!** Someone from the documentation team will review your pull request shortly. They may ask for changes, so keep an eye on your GitHub notifications.
9. If everything looks good, your work will be merged into the `master` branch and become part of the official documentation site. You are now a MeiliSearch Contributor! 🚀

## How We Review Contributions

For **reviewing issues**, we consider a couple criteria:

> 1. Is this task a priority for the documentation team?

Author: I'm not sure how you'd like me to change this.

> 1. Is this task a priority for the documentation team?

Author: How about this:
```suggestion
1. Is this task a priority for our documentation team? Our resources are limited: if it's something we won't do, we always welcome members of the community to implement a fix.
```

2. Is our documentation the best place for this information?

> 3. Is this the first time someone has asked for this?

Author: Good point, plus the next sentence kind of covers it.

## How to Contribute?

### Issues

On the MeiliSearch documentation team, [GitHub Issues](https://github.com/meilisearch/documentation/issues/new/choose) are the primary tool we use to track tasks. They're like elements in our to-do list. Here are some good uses for issues:

- A detailed bug report
- A request for a new technical feature (e.g. dark mode, versioning, etc.)
- A request for new content (e.g. a tutorial, guide, or explanation)
- A helpful suggestion for the future of the documentation

We love issues at MeiliSearch, because they help us do our jobs better. Nine times out of ten, the most useful contribution is a simple GitHub issue that points out a problem and proposes a solution.

#### Creating Your First Issue

All you need to create an issue is a [GitHub account](https://github.com).

1. Log into your account.
2. Go to the MeiliSearch [Documentation repository](https://github.com/meilisearch/documentation).
3. Click on "Issues".
4. Use the search bar to make sure nobody else has reported your issue already. If somebody has, give them an upvote 👍 and **don't create that issue**!
5. Click "New issue".

> 6. Fill in a short, descriptive title and longer summary of your problem or suggestion. If you're reporting a bug, make sure to include steps to reproduce the error, as well as your OS and browser version.

Reviewer: Later you could provide templates for issues, so that when the user clicks on `new issues` he will be met with a default content the same way MeiliSearch and `meilisearch-python` have:
- https://github.com/meilisearch/meilisearch-python/pull/247/files
- https://github.com/meilisearch/MeiliSearch/tree/main/.github/ISSUE_TEMPLATE

2. If you're not already signed in, do so now. You may be prompted to create a [fork](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo)—if so, do it.
3. Use GitHub's text editor to make the change you want.
4. Scroll down to where it says `Propose changes`. In the first text box, write a short description (e.g. "Fix typo"). In the second text box, write a longer description of your changes if necessary.
5. Click `Propose changes` to continue. You should see a page that says `Comparing changes`. This screen is prompting you to create a pull request.
6. Make sure the base repository is `meilisearch/documentation` and the base is `master`. Don't worry about the other two boxes—they automatically target the changes you just made.
7. Below, you can see the changes you made compared to the master branch (what's currently published on docs.meilisearch.com). When you're ready, click `Create pull request`.
8. **Congrats, you made your first PR!** Someone from the documentation team will review your pull request shortly. They may ask for changes, so keep an eye on your GitHub notifications.

> 9. If everything looks good, your work will be merged into the `master` branch and become part of the official documentation site. You are now a MeiliSearch Contributor! 🚀

Reviewer: 😊🔥 so well explained
